### ISSUE TITLE
Add support for rounded corners

### ISSUE THREAD

**Issue by orosbogdan** [rds1983/Myra#472]
Support for rounded corners especially for buttons would improve the UI/UX experience.

**Comment by stevinz** [comment]
This would be nice if it was built in, but I just thought I'd mention that it is already possible to implement yourself thanks to the support for custom brushes.

You can write your own brush class that inherits from [`IBrush`](https://github.com/rds1983/Myra/blob/master/src/Myra/Graphics2D/IBrush.cs). You just need to override the `Draw` function to draw a rounded rectangle. I recommend creating a 9-slice factory with it's own draw function. Then assign an instance of your custom brush to a widget's `Background` / `OverBackground` / `PressedBackground` / etc...

**Comment by sachssoft** [comment]
Definitely, this is not intended and wouldn’t make sense. Just create your own style, and a single TextureAtlas is enough.

**Comment by rds1983** [comment]
btw, what's problem with creating ordinary NinePatchRegion and using it as brush?

**Comment by stevinz** [comment]
> btw, what's problem with creating ordinary NinePatchRegion and using it as brush?

Thanks for pointing out `NinePatchRegion`! In my particular implementation, my brushes generate the rounded textures in code so I can adjust radius, borders, and gradients, but the built-in class looks like a good fit for handling the actual nine-slice drawing.
